### PR TITLE
contrib: update release guide with helm/ansible operator and website info

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
* docs/contribution-guidelines/release.md: add info on website changes and ansible/helm operator release binaries

**Motivation for the change:** new release steps are necessary to document.

/cc @asmacdo @camilamacedo86 @varshaprasad96 

/kind documentation

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: website/content/en/docs/contribution-guidelines/release.md [hunk 1]
```diff
@@ -306,26 +330,17 @@ To github.com:operator-framework/operator-sdk.git
 
 Now that the branch exists, you need to make the post-release PR for the new release branch. To do this, simply follow the same steps as in [step 3](#3-create-a-pr-for-post-release-version-updates) with the addition of changing the branch name in the `go.mod` scaffold from `master` to the new branch (for example, `v1.3.x`). Then, make the PR against the new branch.
 
-### 6. Updating the Homebrew formula
```
Comment (camilamacedo86) [inline website/content/en/docs/contribution-guidelines/release.md:309]: Should we not update the homebrew formula anymore?

**Comment by estroz** [comment]
The website-related steps are incorrect and need re-doing. Specifically, the release PR/tag need to include changes to `website/config.toml`, and the release branch (ex. `v1.3.x`) needs to be created before the release so the website can pick up that release.

/hold

File: website/content/en/docs/contribution-guidelines/release.md [hunk 2]
```diff
@@ -235,25 +242,51 @@ Commit the following changes:
...
+Lock down the master branch to prevent further commits between this and step 4. See [this section](#locking-down-branches) for steps to do so.
+
+Commit the following changes:
+
+- `website/config.toml`: update `version_menu = "Releases"` with the patch-less version string `version_menu = "v1.3"`,
+and add the following lines under `[[params.versions]]` for `master`:
+  ```toml
+  [[params.versions]]
+    version = "v1.3"
+    url = "https://v1-3-x.sdk.operatorframework.io"
```
Comment (asmacdo) [inline website/content/en/docs/contribution-guidelines/release.md:256]: As mentioned [above](https://github.com/operator-framework/operator-sdk/pull/3383/files#diff-c7aebce28b63411bb609012adc8ab83eR186-R187), this link won't work (and will fail CI) unless Netlify is already building this branch. The configuration on the netlify side cannot be done completely ahead of time though, the github branch must be created and pushed first. 

I think if we add a `git push` [here](https://github.com/operator-framework/operator-sdk/pull/3383/files#diff-c7aebce28b63411bb609012adc8ab83eR227) that should cover it.

Comment (estroz) [reply website/content/en/docs/contribution-guidelines/release.md:272]: Done

**Pull request merged**